### DISCUSSION TITLE
Can ble.sh be configured to not show suggestions in the middle of words?

### DISCUSSION THREAD

**Discussion by mattmc3** [akinomyoga/ble.sh#508] [Q&A]
Steps to reproduce:

1. First, I type `cd ~/foo/bar`
2. Next, move the cursor backwards just after the 'b' like so: `cd ~/foo/b|ar`
3. Now, start typing 'a', and ble.sh will complete in the middle the word: `cd ~/foo/ba|STUFF_I_DON'T_WANT_TO_SEEar`

I really only want to see completion suggestions if the cursor is at the end of a line, or when there's whitespace immediately after. So if I type `git clone --depth 1 https://...`, and then go back to start typing another flag `--q` like so: `git clone --q| --depth 1 https://...`, then it's okay to suggest `--quiet` because the cursor is not in the middle of a word. How can I configure ble.sh to behave this way?

**Comment by akinomyoga** [comment]
The `auto-complete` feature is triggered by the function `ble/complete/auto-complete.idle`. You can modify that function using `ble/function#advice` (which I explained in https://github.com/akinomyoga/ble.sh/discussions/490). Since `ble/complete/auto-complete.idle` is defined in `lib/core-complete.sh` which is dynamically loaded, one needs to modify the function through the hook for the complete module:

<details><summary><b>edit</b>: This example doesn't work. See the next reply.</summary>

```
# blerc

function settings-for-completion {
  ble/function#advice around ble/complete/auto-complete.idle '
    if ble/string#match "${_ble_edit_str:_ble_edit_ind}" "^[[:space:]]|^$"; then
      ble/function#advice/do
    fi
  '
}
blehook/eval-after-load complete settings-for-completion
```
</details>

**Reply by akinomyoga** [reply]
Sorry, the above doesn't work. With the above setting, `auto-complete` stops working once it is canceled. One should have called `ble/util/idle.wait-user-input` to tell that the `auto-complete` hook needs to wait until the next user input:

```bash
# blerc

function settings-for-completion {
  ble/function#advice around ble/complete/auto-complete.idle '
    if ble/string#match "${_ble_edit_str:_ble_edit_ind}" "^[[:space:]]|^$"; then
      ble/function#advice/do
    else
      ble/util/idle.wait-user-input
    fi
  '
}
blehook/eval-after-load complete settings-for-completion
```
